crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -89,9 +74,8 @@ impl<T: PartitionProcessorPlacementHints> PartitionProcessorPlacementHints for &
 }
 
 pub struct Scheduler<T> {
-    scheduling_plan: SchedulingPlan,
-    last_updated_scheduling_plan: Instant,
     metadata_store_client: MetadataStoreClient,
+    metadata_writer: MetadataWriter,
```

Comment:
The `metadata_writer` should give you access to the `metadata_store_client`. So we could remove the `metadata_store_client`.